### PR TITLE
fix(Core/Pets): No combat attacking with pet.

### PR DESCRIPTION
**Pull request by HeartWell**
## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2838

## TESTS PERFORMED:
- Tested locally, fixed the issue. Nothing appears to be broken.

## HOW TO TEST THE CHANGES:

1. Make 2 characters, I've tested with a rogue and a frost mage and warlock
2. Attack the rogue with pet and follow the combat.

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by HeartWell** [comment]
Look [https://youtu.be/l3CRjl0DrXQ](video)

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -9447,7 +9447,7 @@ bool Unit::Attack(Unit* victim, bool meleeAttack)
     //if (GetTypeId() == TYPEID_UNIT)
     //    ToCreature()->SetCombatStartPosition(GetPositionX(), GetPositionY(), GetPositionZ());
 
-    if (GetTypeId() == TYPEID_UNIT && !IsPet())
+    if (GetTypeId() == TYPEID_UNIT && !IsPet() && !IsSummon() && IsGuardian())
```
Comment (Helias) [inline src/server/game/Entities/Unit/Unit.cpp:9450]: does it should be `NOT IsGuardian()` ?  so `!IsGuardian()`?

Comment (HeartWell) [reply src/server/game/Entities/Unit/Unit.cpp:9450]: No it's work fine. Look video.

**Comment by r0m1ntik94** [comment]
confirm, tested work fine. Thank @HeartWell

**Comment by Kitzunu** [comment]
Why close?

**Comment by Tyrelis** [comment]
> Why close?

I am testing with and without PR and I will make new pull request.

**Pull request closed without merging**